Development/src/pages/devices/ChannelMappingMatrix.js

```diff
@@ -0,0 +1,514 @@
+                    <CollapseButton
+                        onClick={() => handleEpandCol(output_name)}
+                        isExpand={isColExpanded(output_name)}
+                    />
+                    <Tooltip
+                        title={get_output_caps(output_item, io)}
+                        placement="bottom"
+                    >
+                        <div>{output_item.properties.name}</div>
+                    </Tooltip>
+                </TableCell>
+            ))}
+        </TableRow>
+        <TableRow>
+            {outputs.map(([output_name, output_item]) => {
+                return isColExpanded(output_name)
+                    ? output_item.channels.map(channel => (
+                          <TableCell align="center" size="small">
+                              {channel.label}
+                          </TableCell>
+                      ))
+                    : null;
+            })}
+        </TableRow>
+    </>
+);
+
+const InputsRows = ({
+    inputs,
+    outputs,
+    isColExpanded,
+    isRowExpanded,
+    handleEpandRow,
+    is_show,
+    handleMap,
+    isMapped,
+}) => {
+    return inputs.map(([input_name, input_item]) => (
+        <>
+            <TableRow>
+                <InputSourceAssociation
+                    isRowExpanded={isRowExpanded(input_name)}
+                    input_item={input_item}
+                />
+                <TableCell
+                    size="small"
+                    rowSpan={
+                        isRowExpanded(input_name)
+                            ? input_item.channels.length + 1
+                            : 1
+                    }
+                    colSpan={isRowExpanded(input_name) ? 1 : 2}
+                >
+                    <CollapseButton
+                        onClick={() => handleEpandRow(input_name)}
+                        isExpand={isRowExpanded(input_name)}
+                    />
+                    <Tooltip
+                        title={get_input_caps(input_item)}
+                        placement="bottom"
+                    >
+                        <div>{input_item.properties.name}</div>
+                    </Tooltip>
+                </TableCell>
+                {!isRowExpanded(input_name) && (
+                    <FillEmptyCellsForCollapseRow
+                        outputs={outputs}
+                        isColExpanded={isColExpanded}
+                    />
+                )}
+            </TableRow>
+            {isRowExpanded(input_name) && (
+                <InputChannelsMappingRows
+                    input_item={input_item}
+                    input_nmos_name={input_name}
+                    outputs={outputs}
+                    isColExpanded={isColExpanded}
+                    mapping_disabeld={is_show}
+                    handleMap={handleMap}
+                    isMapped={isMapped}
+                />
+            )}
+        </>
+    ));
+};
+
+const SortByIOName = io_object => {
```

Comment:
I'm not expert on React coding style guidelines, but this looks like a function not React element, so I'd have expected lowercase first letter, e.g. `sortByIOName` in this case?